tests/functional/conftest.py

```diff
@@ -48,10 +55,31 @@
 # Directory storing test-specific data.
 _DATA_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data')
 
-# Provide test access to _DATA_DIR via a fixture.
+# Code
+# ====
+# Fixtures
+# --------
+# Define a fixure for the DataDir object.
 @pytest.fixture
-def data_dir():
-    return _DATA_DIR
+def data_dir(
+  # The request object for this test. See
+  # https://pytest.org/latest/builtin.html#_pytest.python.FixtureRequest
+  # and
+  # https://pytest.org/latest/fixture.html#fixtures-can-introspect-the-requesting-test-context.
+  request,
+  # The tmpdir object for this test. See
+  # https://pytest.org/latest/tmpdir.html.
+  tmpdir):
+
+    # Strip the leading 'test_' from the test's name.
+    name = request.function.__name__[5:]
```

Comment:
Good point. I haven't looked at the old test-suite at all. It might be simple to just put those shared packages and modules into `tests/functional/modules` where we keep all the other shared modules. (This folder is indeed added to `pathex` in the `pyi_builder`)